### DISCUSSION TITLE
remove slug automatically in arabic and persian

### DISCUSSION THREAD

**Discussion by amirajlo** [yiisoft/yii2#19695] [Q&A] [answered]
Whene use slug in Url manager in english,Yii remove automatically slug word from url 
but in arabic and persian language does not delete it.
how to fix it ?

**Comment by samdark** [comment]
Would you please provide a few examples?

**Reply by amirajlo** [reply]
In english:
Labteb.com/tag/hot-plate-magnet

In persian:
https://labteb.com/tag/show?slug=هات-پلیت-و-مگنت

**Reply by bizley** [reply]
This is because of the URL rule regex you have for this route - if you have something like `[\w]+` it will cover only a-z A-Z 0-9 _ . For Persian apparently `[\u0600-\u06FF\s]+` should work but I haven't tested it.

**Reply by amirajlo** [reply]
hi @bizley 
I use `[\u0600-\u06FF\s]+` and is not work.
buy use blow code and work:

`'tag/<slug:[۰۱۲۳۴۵۶۷۸۹ءاآؤئبپتثجچحخدذرزژسشصضطظعغفقکكگگلمنوهیa-za-z0-9-]+>' => 'tag/show',`

this code work in Persian and Arabic language.
Thank You 
Good Luck

**Comment by amirajlo** [comment]
excuse me 
after solve Persian slug 
I have new problem

below link is work
https://labteb.com/tag/هات-پلیت-و-مگنت

but my link index in google like:
https://labteb.com/tag/show?slug=هات-پلیت-و-مگنت
when click on link display error 404 page not found
can I solve this? if not
how to redirect to correct URL when user click this, before display error? like Wordpress 404 plugins

**Comment by amirajlo** [comment, marked as answer]
set below code in rule of urlManager

`'tag/<slug:[۰۱۲۳۴۵۶۷۸۹ءاآؤئبپتثجچحخدذرزژسشصضطظعغفقکكگگلمنوهیa-za-z0-9-]+>' => 'tag/show', `
